joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala

```diff
@@ -1945,4 +1936,37 @@ class AstCreator(filename: String, global: Global)
   def astForYieldWithOptionalArgumentPrimaryContext(ctx: YieldWithOptionalArgumentPrimaryContext): Seq[Ast] = {
     astForYieldWithOptionalArgumentContext(ctx.yieldWithOptionalArgument())
   }
+
+  def resolveRequireOrLoadPath(argsAst: Seq[Ast]): Seq[Ast] = {
+    val importedNode =
+      argsAst.head.nodes.filter(node => node.isInstanceOf[NewLiteral]).head.asInstanceOf[NewLiteral]
```

Comment:
I would suggest being a lot more defensive here, e.g. 
```
argsAst.head.nodes.collect[Literal].headOption
```

Reply:
Yes using the `collectFirst` instead.